app_dart/lib/src/request_handlers/flaky_handler_utils.dart

```diff
@@ -446,21 +448,25 @@ BuilderType getTypeForBuilder(String? builderName, YamlMap ci) {
   return hasFrameworkTag && hasHostOnlyTag ? BuilderType.frameworkHostOnly : BuilderType.unknown;
 }
 
-List<dynamic>? _getTags(String? builderName, YamlMap ci) {
-  final YamlList targets = ci[kCiYamlTargetsKey] as YamlList;
-  final YamlMap? target = targets.firstWhere(
-    (dynamic element) => element[kCiYamlTargetNameKey] == builderName,
-    orElse: () => null,
-  ) as YamlMap?;
+List<String>? _getTags(String? targetName, CiYaml ciYaml, {bool unfilteredTargets = false}) {
+  List<Target> allTargets;
+  if (!unfilteredTargets) {
+    allTargets = ciYaml.presubmitTargets;
+    allTargets.addAll(ciYaml.postsubmitTargets);
```

Comment:
These can overlap (i.e. target A can exist twice in this list). It's cheap to run, but it may be good to make this a `Set` or document the inefficiency

Reply:
Ah okay let me take a closer look at this. Set might be the way to go.

Reply:
Okay made this a Set. I don't want to introduce an inefficiency for sure.